bbb_presentation_video/renderer/tldraw/shape/text.py

```diff
@@ -95,18 +94,22 @@ def finalize_label(
     layout = Pango.Layout(pctx)
     layout.set_auto_dir(True)
     layout.set_font_description(font)
-    if shape.size.width > 0:
-        layout.set_width(int(shape.size.width * Pango.SCALE))
-    if shape.size.height > 0:
-        layout.set_height(int(shape.size.height * Pango.SCALE))
-    layout.set_wrap(Pango.WrapMode.WORD_CHAR)
+    layout.set_line_spacing(0.4)
     layout.set_alignment(Pango.Alignment.CENTER)
 
     layout.set_text(shape.label, -1)
 
-    (_, layout_height) = layout.get_pixel_size()
-    height_offset = (shape.size.height - layout_height) / 2
-    ctx.translate(0, height_offset)
+    (layout_width, layout_height) = layout.get_pixel_size()
+    if style.dash is DashStyle.DRAW or isinstance(shape, TriangleShape):
+        width_offset = (shape.size.width - layout_width) / 2 
+        height_offset = (shape.size.height - layout_height) / 2
+    else:
+        width_offset = (-layout_width) / 2
+        height_offset = (-layout_height) / 2
+    # label of triangle is not exactly centered   
+    if (isinstance(shape, TriangleShape)):
```

Comment:
This seems very odd, does the tldraw code also have special handling for offsetting labels of triangle in the renderer? I don't recall seeing any.

Reply:
[It has](https://github.com/tldraw/tldraw/blob/main/packages/tldraw/src/state/shapes/TriangleUtil/TriangleUtil.tsx#L82-L86). I copied it now to be the same.